upgrade.sh

```diff
@@ -52,6 +52,66 @@ function unbound_psp_check() {
+    alert_pv=$(kubectl get pvc -n sysmgmt-health --output=json $alert_pvc |jq '.spec."volumeName"')
+    prom_pv="${prom_pv//[\",]}"
+    alert_pv="${alert_pv//[\",]}"
+    echo "$prom_pv"
+    echo "$alert_pv"
+
+    # Patch the PersistenceVolume created/used by the prometheus-operator and alertmanager to Retain claim policy
+    kubectl patch pv/$prom_pv -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}'
+    kubectl patch pv/$alert_pv -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}'
+
+    # Uninstall the cray-sysmgmt-health release
+    helm ls -o json --namespace sysmgmt-health| jq -r --argjson version $version '.[] | select(.app_version | sub(".[0-9]$";"") | tonumber | . < $version).name'| xargs -L1 helm uninstall --namespace sysmgmt-health
+    
+    # Delete the existing PersistentVolumeClaim, and verify PV become Released
+    kubectl delete pvc/$prom_pvc -n sysmgmt-health
+    kubectl delete pvc/$alert_pvc -n sysmgmt-health
+    alert_pv_phase=$(kubectl get pv $prom_pv -o json |jq '.status."phase"')
+    prom_pv_phase=$(kubectl get pv $alert_pv -o json |jq '.status."phase"')
+    prom_pv_phase="${prom_pv_phase//[\",]}"
+    alert_pv_phase="${alert_pv_phase//[\",]}"
+    echo "Verifying whether PVs became Released or not."
+    if [ "$alert_pv_phase" == Released ] && [ "$prom_pv_phase" == Released ]; then
+        echo "Both Prometheus and Alertmanager PVs are Released"
+    else
+        echo >&2 "PVs are not Released. Verify if PV exists or not."
+    fi
+    # Remove the cray-sysmgmt-health-promet-kubelet service.
+    echo "Deleting cray-sysmgmt-health-promet-kubelet service in kube-system namespace."
+    kubectl delete service/cray-sysmgmt-health-promet-kubelet -n kube-system
+
+    echo "Deleting sysmgmt-health existing CRDs"
+    # Remove all the existing CRDs (ServiceMonitors, Podmonitors, etc.)
+    kubectl delete crd alertmanagers.monitoring.coreos.com podmonitors.monitoring.coreos.com prometheuses.monitoring.coreos.com prometheusrules.monitoring.coreos.com servicemonitors.monitoring.coreos.com thanosrulers.monitoring.coreos.com
+
+    # Remove current spec.claimRef values to change the PV's status from Released to Available.
+    kubectl patch pv/$prom_pv --type json -p='[{"op": "remove", "path": "/spec/claimRef"}]'
+    kubectl patch pv/$alert_pv --type json -p='[{"op": "remove", "path": "/spec/claimRef"}]'
+    alert_pv_phase=$(kubectl get pv $prom_pv -o json |jq '.status."phase"')
+    prom_pv_phase=$(kubectl get pv $alert_pv -o json |jq '.status."phase"')
+    prom_pv_phase="${prom_pv_phase//[\",]}"
+    alert_pv_phase="${alert_pv_phase//[\",]}"
+    echo "Verifying whether PV became Available or not."
+    if [ "$alert_pv_phase" == Available ] && [ "$prom_pv_phase" == Available ]; then
+        echo "Both Prometheus and Alertmanager PVs are Available. Ready to deploy the latest cray-sysmgmt-chart now."
+    else
+        echo >&2 "PVs are not Available. Verify if PV exists or not."
```

Comment:
Same thing here as above both with a loop with a pause might not cause timing issues as well as the else case should probably indicate a failure condition for the script.

Would also be useful to show the pv names in the error cases so that people know what to look for.